activerecord/test/cases/enum_test.rb

```diff
@@ -1006,4 +1006,17 @@ def self.name
   ensure
     ActiveRecord::Base.logger = old_logger
   end
+
+  test "raises for columnless enums" do
+    klass = Class.new(ActiveRecord::Base) do
+      def self.name
+        "Book"
+      end
+      enum columnless_genre: [:adventure, :comic]
+    end
+
+    assert_raises(ActiveRecord::UnknownAttributeError) do
```

Comment:
Would be good to assert on the error message, I'd like to see if it's somewhat actionable given that the backtrace won't point to the source of the error.

Reply:
Good idea. We can not use `ActiveRecord::UnknownAttributeError` here, because it is for use with the actual records. So changed this too.